### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by MrPetovan**
This PR was originally meant to fix the same notices as #9721 did, and now that they are already addressed, here are the remaining "side improvement" I added while I was working on the notices.

### CODE REVIEW

File: mod/events.php [hunk 1]
```diff
@@ -150,26 +150,16 @@ function events_post(App $a)
...
-	$c = q("SELECT `id` FROM `contact` WHERE `uid` = %d AND `self` LIMIT 1",
-		intval(local_user())
-	);
-
-	if (DBA::isResult($c)) {
-		$self = $c[0]['id'];
-	} else {
-		$self = 0;
-	}
+	$self = \Friendica\Model\Contact::getPublicIdByUserId($uid);
```
Comment (annando) [inline mod/events.php:153]: The function `getPublicIdByUserId` does return the public contact id of the self contact. The result is not the same as the result of the replaced query.

**Review by annando — COMMENTED** [review]
I'm unsure about this changed function.

File: mod/events.php (hunk 1 above)
Comment (MrPetovan) [reply mod/events.php:153]: This query is used to populate `$self` which is only used to generate self-only permission string (in lines 176 and 179) which should use the public contact id of the local user. I don't even know how it was working before with the private self contact id.

Comment (annando) [reply mod/events.php:153]: AFAIK the ACL always worked with the user's contact ids. This is something that I have to check against before being able to merge it.

Comment (MrPetovan) [reply mod/events.php:153]: We use public contact ids for the rest of the permissions, I believe this is a good normalization change.

Comment (annando) [reply mod/events.php:153]: Looking at https://github.com/friendica/friendica/blob/2020.12-rc/src/Repository/PermissionSet.php#L163-L182 it seems that we expect the user's contact ids. And looking at the way the function is called: https://github.com/friendica/friendica/blob/2020.12-rc/src/Protocol/ActivityPub/Transmitter.php#L248-L250 or https://github.com/friendica/friendica/blob/2020.12-rc/src/Module/Profile/Status.php#L219-L221 or https://github.com/friendica/friendica/blob/2020.12-rc/src/Module/Objects.php#L83-L85 it seems as if we are expected to use the user's contact id.

Comment (MrPetovan) [reply mod/events.php:153]: The public contact id or the self contact id?

Comment (annando) [reply mod/events.php:153]: AFAIK (when looking at the links) we always provide the user's contact id to the ACL fetching when reading posts. So I guess we should only feed the ACL with the user's contact id as well.

Comment (MrPetovan) [reply mod/events.php:153]: Again, the public contact id or the self contact id? Both can be described as the "user's contact id".

Comment (annando) [reply mod/events.php:153]: The public id always belongs to the "uid=0", the self contact always belongs to the user, so the self contact is the user's contact id.

I just created a PR to handle this: https://github.com/friendica/friendica/pull/9732

Comment (MrPetovan) [reply mod/events.php:153]: Thank you for the elaboration, looking forward to the end of this sort of confusion.

**Pull request merged**